### PR TITLE
More typescript porting

### PR DESCRIPTION
**Pull request by JJL772**
Ports the main menu script to typescript, bumps types submodule and fixes the controls library.

Just playing around with typescript mainly, please review so I can learn!

### CODE REVIEW

File: scripts/pages/main-menu/main-menu.ts [hunk 1]
```diff
@@ -2,28 +2,16 @@
...
-		/** @type {Image} @static */
-		image: $('#MainMenuBackground'),
-		/** @type {ModelPanel} @static */
-		model: $('#MainMenuModel'),
-		/** @type {Image} @static */
-		mapSelectorBackground: $('#MainMenuBackgroundMapSelectorImage'),
-		/** @type {Panel} @static */
+		movie: $('#MainMenuMovie') as Movie,
+		image: $('#MainMenuBackground') as Image,
+		model: $('#MainMenuModel') as ModelPanel,
```
Comment (craftablescience) [inline scripts/pages/main-menu/main-menu.ts:12]: Use the C++ style template syntax that Baguettery added for stuff like this (there's more places but not gonna comment on all of them)
```suggestion
		movie: $<Movie>('#MainMenuMovie'),
		image: $<Image>('#MainMenuBackground'),
		model: $<ModelPanel>('#MainMenuModel'),
```

**Review by craftablescience — CHANGES_REQUESTED** [review]
Everything else looks good

File: scripts/pages/main-menu/main-menu.ts (hunk 1 above)
Comment (JJL772) [reply scripts/pages/main-menu/main-menu.ts:12]: Cool, this is much nicer.

Comment (JJL772) [reply scripts/pages/main-menu/main-menu.ts]: Ok applied this.

**Review by craftablescience — APPROVED** [review]

**Pull request merged**